packages/components/src/IDReader/index.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import React from 'react'
+import { Box } from '../Box'
+import styled from 'styled-components'
+import { Stack } from '../Stack'
+import { Divider } from '../Divider'
+import { Text } from '../Text'
+import { QRReader } from './readers/QRReader/QRReader'
+import { ReaderType, ScannableIDReader } from './types'
+
+const StyledBox = styled(Box)`
+  background: ${({ theme }) => theme.colors.background};
+`
+
+export const IDReader = (props: ScannableIDReader) => {
+  const {
+    dividerLabel,
+    manualInputInstructionLabel,
+    readers,
+    onScan,
+    onError
+  } = props
+  const renderReader = (reader: ReaderType) => {
+    const { type, ...readerProps } = reader
+    if (type === 'qr') {
+      return (
+        <QRReader
+          key={type}
+          {...readerProps}
+          onScan={onScan}
+          onError={onError}
+        />
+      )
+    } else return null
+  }
```

Comment:
Basically this is a "ReaderGenerator" (like FormFieldGenerator), can be lifted to a separate component so it doesn't get re-created on every render of IDReader

Reply:
Agreed, I am on it 👍🏼